core/arch/arm/sm/sm_a32.S

```diff
@@ -216,6 +216,28 @@ FUNC sm_init , :
 	mrs	r1, cpsr
 	cps	#CPSR_MODE_MON
 	mov	sp, r0
+
```

Comment:
It can be done in hyp mode also.

If you don't want to do this in hyp mode after OP-TEE is initialized I suggest doing it in mon mode before initializing OP-TEE.